src/app/reset-password/actions.ts

```diff
@@ -0,0 +1,79 @@
+'use server'
+
+import { CustomerResetByUrlPayload } from '@shopify/hydrogen-react/storefront-api-types'
+import { redirect } from 'next/navigation'
+import { ServerActionError } from '@/@types/common'
+import { ResetPasswordForm, ResetPasswordFormSchema } from '@/@types/customer'
+import Customers from '@/database/dtos/customers'
+import storefrontApi from '@/service/api/storefrontApi'
+import { Route } from '@/utils/constants/routes'
+import {
+  StorefrontDataKey,
+  StorefrontErrorKey,
+} from '@/utils/constants/storefrontGql'
+import {
+  handleStorefrontGqlResponse,
+  isZodError,
+} from '@/utils/functions/common'
+import { logger } from '@/utils/functions/logger'
+import { getPasswordHash } from '@/utils/functions/password'
+
+export async function resetPassword(
+  _: undefined | ServerActionError<ResetPasswordForm>,
+  form: FormData,
+): Promise<undefined | ServerActionError<ResetPasswordForm>> {
+  const data = {
+    password: form.get('password')?.toString() || '',
+    verifyPassword: form.get('verifyPassword')?.toString() || '',
```

Comment:
do you use this in the action? i thought is usually client side check if both are the same